src/App.tsx

```diff
@@ -1,35 +1,178 @@
+import '@fortawesome/fontawesome-free/css/all.css';
+import 'bulma';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { useState } from 'react';
+import cn from 'classnames';
+
+type User = {
+  id: number;
+  name: string;
+  username: string;
+  email: string;
+};
+
+type Todo = {
+  id: number;
+  title: string;
+  completed: boolean;
+  userId: number;
+  user: User | null;
+};
+
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+// const zeroForValueOptionSelect: number = 0;
+
+function getNewTodoId(todoList: Todo[]) {
+  const maxId = Math.max(...todoList.map(todo => todo.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [title, setTitle] = useState<string>('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState<number>(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+  const [currentTodos, setCurrentTodos] = useState(todos);
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
```

Comment:
The `handleSubmit` function correctly prevents form submission if the title or user ID is missing. Consider adding user feedback for successful todo addition, such as a confirmation message or toast notification.